ideascube/monitoring/forms.py

```diff
@@ -143,14 +155,13 @@ class ReturnForm(forms.Form):
 
     def clean_loan(self):
         barcode = self.cleaned_data['loan']
-        try:
-            loan = Loan.objects.due().get(specimen__barcode=barcode)
-        except Loan.DoesNotExist:
-            msg = _('Item with barcode {barcode} is not loaned.')
+        loan = Loan.objects.due().filter(specimen__barcode=barcode).first()
+        if loan is None:
+            loan = Loan.objects.due().filter(specimen__serial=barcode).first()
+        if loan is None:
+            msg = _('Item with barcode or serial number {barcode} is not '
+                    'loaned.')
```

Comment:
Because this (previous) `try/except` except for `Loan.DoesNotExist` but also `Loan.MultipleObjectsReturned`.
Adding the same code (one try and two except) in the `Loan.DoesNotExist` execpt clause seems a bit to big for me.